src/town_cmd.cpp

```diff
@@ -905,6 +905,70 @@ static bool IsRoadAllowedHere(Town *t, TileIndex tile, DiagDirection dir)
 	return ret;
 }
 
+/**
+ * Check whether growing on a half-tile coast tile ends up blocking a water connection
+ *
+ * @param tile The target tile
+ * @return true if building here blocks a water connection
+ */
+static bool GrowingOnWateredTile(TileIndex tile)
```

Comment:
I think the function name could contain more information about what it does. Taking words from the comment a name like GrowingBlocksWaterConnection might be better. This way you can understand immediately when should it return with true or false.